akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -237,12 +237,13 @@ object ZipWith extends ZipWithApply
  * '''Cancels when''' any downstream cancels
  */
 object Unzip {
+
   /**
    * Create a new `Unzip`.
    */
   def apply[A, B](): Unzip[A, B] = {
     val shape = new FanOutShape2[(A, B), A, B]("Unzip")
-    new Unzip(shape, new UnzipModule(shape, OperationAttributes.name("Unzip")))
+    new Unzip(shape, new UnzipWith2Module[(A, B), A, B](shape, identity, OperationAttributes.name("Unzip")))
```

Comment:
This aims to be the follow up to https://github.com/akka/akka/pull/17533/files#r30879038 
It is not yet exactly what was meant there - by using a `def identity` which the `Predef` identity is ETA expansion happens on this line, which means that a new `Function1` object will be allocated each time we hit this line.
The solution to save on these allocations is to:

```scala
object Unzip {
  private final val _identity: Any => Any = a => a

  // .. 

  def apply[...] = { ... 
  new Unzip(shape, new UnzipWith2Module[(A, B), A, B](
    shape, 
    identity.asInstanceOf[(A, B) => (A, B)], 
    OperationAttributes.name("Unzip")))
```

// mini sanity check:

```
scala> def take(i: Int, id: Int => Int) = id.##
take: (i: Int, id: Int => Int)Int

scala> take(2, identity)
res10: Int = 1910438136

scala> take(2, identity)
res11: Int = 914507705

scala> take(2, ident.asInstanceOf[Int => Int])
res13: Int = 1325124186

scala> take(2, ident.asInstanceOf[Int => Int])
res14: Int = 1325124186
```

This way we save on allocations which can hurt performance :)